lib/job.js

```diff
@@ -137,21 +142,19 @@ class Job extends BaseModel {
      * @return {Promise}        List of running builds
      */
     getRunningBuilds() {
-        const listConfig = {
-            params: {
-                jobId: this.id,
-                status: ['RUNNING', 'QUEUED']
-            }
-        };
-
-        // Lazy load factory dependency to prevent circular dependency issues
-        // https://nodejs.org/api/modules.html#modules_cycles
-        /* eslint-disable global-require */
-        const BuildFactory = require('./buildFactory');
-        /* eslint-enable global-require */
-        const factory = BuildFactory.getInstance();
+        return Promise.all([
+            this.getBuilds({ status: 'RUNNING' }),
+            this.getBuilds({ status: 'QUEUED' })
+        ]).then(([runningBuilds, queuedBuilds]) => [...runningBuilds, ...queuedBuilds]);
+    }
 
-        return factory.list(listConfig);
+    /**
+     * Return the last successful build that belong to this job
+     * @return {Promise}        Resolves to the last successful build of this job
+     */
+    getLastSuccessfulBuild() {
+        return this.getBuilds({ status: 'SUCCESS' })
```

Comment:
i guess this queries DB in background, can we modify it so that count is addressed as part of db query and not in code afterwards ?

Reply:
Pagination is not implemented yet: https://github.com/screwdriver-cd/screwdriver/issues/233

Reply:
We should look at fixing it soon. Not comfortable with querying DB for all records for each API call.

Reply:
@jithin1987  Sure. Do you want to push that up higher in the priority?